[Swift in WebKit] The Swift implementation of intelligence text effects code is slightly incorrect

rr-codes opened the pull request:
#### 1243db2d3cef54f8258b2ace6c8dd71476efe8c7
<pre>
[Swift in WebKit] The Swift implementation of intelligence text effects code is slightly incorrect
<a href="https://bugs.webkit.org/show_bug.cgi?id=293856">https://bugs.webkit.org/show_bug.cgi?id=293856</a>
<a href="https://rdar.apple.com/152373241">rdar://152373241</a>

Reviewed by Aditya Keerthi.

Properly use `@objc @implementation` and address all the errors it exposes.

Also apply formatting and linting.

* Source/WebKit/WebKitSwift/WritingTools/IntelligenceTextEffectViewManager.swift:
* Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceReplacementTextEffectCoordinator.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceSmartReplyTextEffectCoordinator.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceTextEffectCoordinator.h:

- Format code
- Remove needless `public`
- Replace force-unwrap with optional-unwrap
- Justify other force-unwraps

Canonical link: <a href="https://commits.webkit.org/295726@main">https://commits.webkit.org/295726@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1858c7a14effececd615282fa71a78a0da96f2e8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105965 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25715 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16109 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111162 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56562 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108005 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26355 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34218 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80497 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108970 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/20777 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95625 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60817 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/20393 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13722 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56000 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90158 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13758 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114014 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33104 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/24425 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89569 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33468 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91853 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89249 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22748 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34112 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11926 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28643 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33029 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38440 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32775 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36124 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34373 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift:
```diff
@@ -585,6 +622,8 @@ struct PlatformIntelligenceTextEffectID: Hashable {
             // This block is invoked after the source effect has been prepared, but before it actually begins.
             // It's intended for the destination effect to be added here, synchronously.
 
+            // Mis-annotated WritingToolsUI symbol which never actually returns `nil`.
```
```suggestion
            // Misannotated WritingToolsUI symbol which never actually returns `nil`.
```

pxlcoder commented on Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift:
```diff
@@ -597,28 +636,31 @@ struct PlatformIntelligenceTextEffectID: Hashable {
 
         await view.source.replacementEffectWillBegin(self)
 
+        // Mis-annotated WritingToolsUI symbol which never actually returns `nil`.
```
```suggestion
        // Misannotated WritingToolsUI symbol which never actually returns `nil`.
```

pxlcoder commented on Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceReplacementTextEffectCoordinator.swift:
```diff
@@ -145,38 +199,64 @@ import WebKitSwift
-    @objc(showEffectsWithCompletion:)
-    public func showEffects() async {
+    func showEffects(completion: @MainActor @Sendable @escaping () -> Void) {
+        Task {
```
Please explain the introduction of `Task` here and elsewhere in the commit message.

rr-codes committed:
[Swift in WebKit] The Swift implementation of intelligence text effects code is slightly incorrect
https://bugs.webkit.org/show_bug.cgi?id=293856
rdar://152373241

Reviewed by Aditya Keerthi.

Properly use `@objc @implementation` and address all the errors it exposes.

Also apply formatting and linting.

* Source/WebKit/WebKitSwift/WritingTools/IntelligenceTextEffectViewManager.swift:
* Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceReplacementTextEffectCoordinator.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceSmartReplyTextEffectCoordinator.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceTextEffectCoordinator.h:

- Format code
- Remove needless `public`
- Replace force-unwrap with optional-unwrap
- Justify other force-unwraps

Canonical link: https://commits.webkit.org/295726@main
